protocol/nats_jetstream/v2/receiver.go

```diff
@@ -81,6 +104,26 @@ func NewConsumer(url, stream, subject string, natsOpts []nats.Option, jsmOpts []
 	return c, err
 }
 
+// NewConsumerV2 consumes from filterSubjects given in ConsumerConfig or OrderedConsumerConfig
+// embedded in ConsumerOption.
+// See: WithConsumerConfig(...) and WithOrderedConsumerConfig(...)
+func NewConsumerV2(url string, natsOpts []nats.Option, jsOpts []jetstream.JetStreamOpt, opts ...ConsumerOption) (*Consumer, error) {
```

Comment:
Question: do those need to be public?

Reply:
I assume you mean the options?  Maybe possible to drop jetstream.JetStreamOpt

nats.Option:
About 40 to 50 options

jetstream.JetStreamOpt:
1. WithClientTrace(ct *ClientTrace) JetStreamOpt
2. WithPublishAsyncErrHandler(cb MsgErrHandler) JetStreamOpt 
3. WithPublishAsyncMaxPending(max int) JetStreamOpt 

The ConsumerOption options are needed because there are variations of how the NATS consumer is created. 
There are two variations, "normal" and "ordered".  Each has mutiple fields on how to configure.